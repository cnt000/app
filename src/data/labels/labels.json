{
  "openMenuLabel": "apri il menu",
  "title": "Pungilandia",
  "titleDescription": "Piante grasse, strane ed insolite",
  "goToCartLabel": "vai al carrello",
  "copyright": "Copyright © 2020",
  "privacy": "privacy",
  "contact": "contatti",
  "company": "Azienda Agricola Pungilandia Rocchi Domenico",
  "companyAddress": "Via Isonzo 11, 65123 Pescara P.IVA 01203140684 CF RCCDNC59R30I712B\nIscrizione REA Pescara Nr.97422",
  "disclaimer": "Azienda in possesso Registro di Detenzione di Specie Vegetali Incluse\nNegli Allegati CITES A e B del Regolamento (CE) 338/97 del Consiglio del 9\nDicembre 1996 e Successive Modifiche e Attuazioni. Registro N.000001/2001\nVAB",
  "mailToLink": "mailto:info@pungilandia.com",
  "privacyLink": "http://www.pungilandia.com",
  "addToCart": "Aggiungi al carrello",
  "searchPlants": "Cerca piante",
  "searchPlantsImage": "http://www.pungilandia.com/php/images/lo/cotyl_mediop.jpg",
  "hintsFromPungilandiaTitle": "I Consigli di Pungilandia",
  "articleBy": "A cura del dott. Mimmo Rocchi",
  "articleContent": "consigliColtivazione.md",
  "mimmoImage": "http://www.pungilandia.com/media/MIMMO.JPG",
  "menuChiSiamo": "Chi siamo",
  "menuListaPiante": "Lista piante",
  "menuNovita": "Novità",
  "menuProdottiSpeciali": "Prodotti speciali",
  "menuConsigliPerColtivare": "Consigli per coltivare",
  "menuOrdini": "Ordini",
  "menuFotoShow": "Foto show",
  "menuEmail": "E-mail"
}
