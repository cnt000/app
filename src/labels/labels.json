{
  "openMenuLabel": "apri il menu",
  "title": "Pungilandia",
  "titleDescription": "Piante grasse, strane ed insolite",
  "goToCartLabel": "vai al carrello",
  "copyright": "Copyright Â© 2020",
  "privacy": "privacy",
  "contact": "contatti",
  "company": "Azienda Agricola Pungilandia Rocchi Domenico",
  "companyAddress": "Via Isonzo 11, 65123 Pescara P.IVA 01203140684 CF RCCDNC59R30I712B<br>Iscrizione REA Pescara Nr.97422",
  "disclaimer": "Azienda in possesso Registro di Detenzione di Specie Vegetali Incluse<br>Negli Allegati CITES A e B del Regolamento (CE) 338/97 del Consiglio del 9<br>Dicembre 1996 e Successive Modifiche e Attuazioni. Registro N.000001/2001<br>VAB"
}
