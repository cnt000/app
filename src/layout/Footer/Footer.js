import React from 'react';
import styles from './Footer.module.css';

const Footer = ({ copyright, privacy, contact, company, companyAddress, disclaimer }) => (
  <footer className={styles.footer}>
    <p>
      <span className={styles.copyright}>{copyright}</span> -
      <a className={styles.privacy} href="http://www.pungilandia.com">
        Privacy
      </a>{' '}
      -
      <a className={styles.contactus} href="mailto:info@pungilandia.com">
        Contatti
      </a>
    </p>
    <address>
      Azienda Agricola Pungilandia Rocchi Domenico
      <br />
      Via Isonzo 11, 65123 Pescara P.IVA 01203140684 CF RCCDNC59R30I712B
      Iscrizione REA Pescara Nr.97422
    </address>
    <p className={styles.laws}>
      Azienda in possesso Registro di Detenzione di Specie Vegetali Incluse
      Negli Allegati CITES A e B del Regolamento (CE) 338/97 del Consiglio del 9
      Dicembre 1996 e Successive Modifiche e Attuazioni. Registro N.000001/2001
      VAB
    </p>
  </footer>
);

export default Footer;
