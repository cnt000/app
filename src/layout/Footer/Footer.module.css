.footer {
  grid-area: footer;
  padding: 0.5rem;
  text-align: center;
}

.contactus {
  font-size: 16px;
  margin-top: 0.4em;
}

.address,
.laws {
  font-size: 12px;
  margin-top: 0.4rem;
  line-height: 1.2;
}

.laws {
  width: 80%;
  margin: 0 auto;
  font-weight: 300;
}

.laws {
  display: none;
}


@media only screen and (min-width: 1025px) {
  .laws {
    display: block;
    margin-top: 0.4rem;
    width: 60%;
  }
}
