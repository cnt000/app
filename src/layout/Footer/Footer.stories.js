import React from 'react';

import Footer from './Footer';
import labels from '../../labels/labels.json';

export default {
  component: Footer,
  title: 'Footer',
};

export const Default = () => <Footer {...labels} />;
