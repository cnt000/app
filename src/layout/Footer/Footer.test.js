import React from 'react';
import renderer from 'react-test-renderer';

import Footer from './Footer';
import { render } from '@testing-library/react';
import labels from '../../data/labels/labels.json';
import LabelsContext from '../../features/Labels';

const footer = (
  <LabelsContext.Provider value={labels}>
    <Footer />
  </LabelsContext.Provider>
);

it('renders correctly', () => {
  const tree = renderer.create(footer).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders P.IVA and CF', () => {
  const { getByText } = render(footer);
  expect(
    getByText(/Azienda Agricola Pungilandia Rocchi Domenico/),
  ).toBeInTheDocument();
  expect(
    getByText(/P.IVA 01203140684 CF RCCDNC59R30I712B/),
  ).toBeInTheDocument();
});
