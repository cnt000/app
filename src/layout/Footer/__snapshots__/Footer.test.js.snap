// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<footer
  className="footer"
>
  <p>
    <span
      className="copyright"
    />
     -
     
    <a
      className="privacy"
    />
     
    -
     
    <a
      className="contactus"
    />
  </p>
  <address>
    <br />
  </address>
  <p
    className="laws"
  />
</footer>
`;
